nextcore/common/dispatcher.py

```diff
@@ -441,22 +443,23 @@ async def dispatch(self, event_name: EventNameT, *args: Any) -> None:
         """
         logger.debug("Dispatching event %s", event_name)
 
-        # Event handlers
-        # Tasks are used here as some event handler/check might take a long time.
-        for handler in self._global_event_handlers:
-            logger.debug("Dispatching to a global handler")
-            create_task(self._run_global_event_handler(handler, event_name, *args))
-        for handler in self._event_handlers.get(event_name, []):
-            logger.debug("Dispatching to a local handler")
-            create_task(self._run_event_handler(handler, event_name, *args))
-
-        # Wait for handlers
-        for check, future in self._wait_for_handlers.get(event_name, []):
-            logger.debug("Dispatching to a wait_for handler")
-            create_task(self._run_wait_for_handler(check, future, event_name, *args))
-        for check, future in self._global_wait_for_handlers:
-            logger.debug("Dispatching to a global wait_for handler")
-            create_task(self._run_global_wait_for_handler(check, future, event_name, *args))
+        async with create_task_group() as tg:
```

Comment:
Wont this block until all callbacks have finished running? Because if so, that is going to cause issues.

Reply:
> When the task encounters an `await` statement that requires the task to sleep until something happens, the event loop is then free to work on another task.
[AnyIO Docs](https://anyio.readthedocs.io/en/3.x/tasks.html)

It looks like it is blocking. I'm not sure what else to do though.

Reply:
Could a per-dispatcher task group be created?